Source/DafnyCore/DafnyCore.csproj

```diff
@@ -19,12 +19,14 @@
     <DefineConstants>TRACE</DefineConstants>
     <TargetFramework>net6.0</TargetFramework>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <NoWarn>$(NoWarn);NU5104</NoWarn>
```

Comment:
Required because System.CommandLine is in beta.

Reply:
Better as a code comment than a PR comment :)

I've used `CommandLineParser` in a couple of other projects (TestDafny and XUnitExtensions), would you mind just adding a comment somewhere in the code base on what we're using in `System.CommandLine` that isn't in that library?

Reply:
> I've used CommandLineParser in a couple of other projects (TestDafny and XUnitExtensions), would you mind just adding a comment somewhere in the code base on what we're using in System.CommandLine that isn't in that library?

O, honestly I haven't tried `CommandLineParser`, but I imagine it would meet our needs. I chose `System.CommandLine` because it seems more futureproof: https://github.com/dotnet/BenchmarkDotNet/issues/1016

One interesting feature that `System.CommandLine` has is automatic completion support: https://learn.microsoft.com/en-us/dotnet/standard/commandline/tab-completion

This PR doesn't document that for Dafny users though..